src/util/Recolor.cpp

```diff
@@ -0,0 +1,52 @@
+#include "util/Recolor.h"
+
+#include "util/Color.h"
+
+Recolor::Recolor(const ColorU8& light, const ColorU8& dark): dark(dark), light(light) { recalcDiffAndOff(); }
+
+const ColorU8& Recolor::getDark() const { return dark; }
+
+const ColorU8& Recolor::getLight() const { return light; }
+
+void Recolor::recalcDiffAndOff() {
+    difference = ColorU8{
+            static_cast<uint8_t>(std::abs(static_cast<int16_t>(dark.red) - static_cast<int16_t>(light.red))),
+            static_cast<uint8_t>(std::abs(static_cast<int16_t>(dark.green) - static_cast<int16_t>(light.green))),
+            static_cast<uint8_t>(std::abs(static_cast<int16_t>(dark.blue) - static_cast<int16_t>(light.blue)))};
+
+    offset = ColorU8{std::min(dark.red, light.red), std::min(dark.green, light.green), std::min(dark.blue, light.blue)};
+}
+
+ColorU8 Recolor::convertColor(const ColorU8& other) const {
+    return ColorU8{
+            static_cast<uint8_t>(std::min<int16_t>(
+                    255, std::min<int16_t>(255, (255 - static_cast<int16_t>(other.red)) *
+                                                                static_cast<int16_t>(difference.red) / 255 +
+                                                        offset.red))),
```

Comment:
actually no, it`s `(255 - col) * diff/255 + offset` and one `min` for the multiplication and one for the addition. Or is your take since we have so much more space in `int16_t`, one `min` at the end suffices?

Reply:
Oh yes, you're completely right, my bad :+1: